### PR TITLE
Test Data Generation I - support generating schemas/tables from query without authentication

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Test Data Generation I - support a new API to generate skeleton schema/table from query

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://github.com/finos/legend-studio/assets/73408381/3cebbd39-a73f-4a37-b51b-4bd37f9621fb

### CODE REVIEW

File: .changeset/brave-kiwis-yell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-application-studio': patch
+---
+
+support a new API to generate schema/table skeleton from query
```
Comment (MauricioUyaguari) [inline .changeset/brave-kiwis-yell.md:5]:
````suggestion
support generating schemas/tables from query without authentication
````

File: packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestDataEditor.tsx [hunk 2]
```diff
@@ -199,6 +201,12 @@ export const ConnectionTestDataEditor = observer(
       }
     };
 
+    const generateSkeletonTestData = (): void => {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestDataEditor.tsx:204]: lets not sure skeleton in the code. 
we can say bare or just generate schemas

`generateSkeletonTestData ` -> `generateQuerySchemas`

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts [hunk 3]
```diff
@@ -156,6 +158,7 @@ export class ConnectionTestDataState {
   readonly connectionData: ConnectionTestData;
   readonly parametersState: ServiceTestDataParametersState;
   readonly generatingTestDataState = ActionState.create();
+  readonly generatingSkeletonTestDataState = ActionState.create();
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts:161]: generateSchemaQueryState

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts [hunk 4]
```diff
@@ -326,6 +331,63 @@ export class ConnectionTestDataState {
     }
   }
 
+  *generateSkeletonTestData(): GeneratorFn<void> {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts:334]: generateQuerySchemas

File: packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts [hunk 5]
```diff
@@ -326,6 +331,63 @@ export class ConnectionTestDataState {
...
+              this.connectionData,
+              testDataGenerationResult.data[0],
+              this.editorStore.changeDetectionState.observerContext,
+            );
+            this.embeddedEditorState = new EmbeddedDataEditorState(
+              this.testDataState.editorStore,
+              this.connectionData.testData,
+            );
+          } else {
+            this.editorStore.applicationStore.notificationService.notifyError(
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts:369]: lets fix the error messaging here.
we should throw in the above for the reason we can't generate. 
so we should do something like
AsserTrue(testDataGenerationResult.length === 1, 'Expected generated data to be of size 1');
also if the runtime is not a pointer, same thing. 

The catch below will be responsible to catch and notify user of the error. you dont need the notify up here in the try logic.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
Small changes to be fixed for merging. 
Main things are:
1. lets remove skeleton naming from code as seems to crude
2. fix erroring messaging in the generateTestData

Other thing, which we can fix after, what is the reason we need runtime to be a runtimePointer?

**Comment by YannanGao-gs** [comment]
> what is the reason we need runtime to be a runtimePointer

Engine needs runtime path

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**